imas_paraview/plugins/profiles_2d.py

```diff
@@ -258,28 +258,32 @@ def _create_vtkscalars(self, profile):
         return vtk_scalars
 
     def _create_ugrid(self, vtk_points, vtk_scalars):
-        """Create a vtkUnstructuredGrid of the given profile.
+        """Create a vtkUnstructuredGrid using VTK_QUAD cells from the structured 2D
+        grid.
 
         Args:
-            profile: The profile to create a ugrid for.
-
-        Returns:
-            The created unstructured grid.
+            vtk_points: The VTK points representing the grid vertices.
+            vtk_scalars: The scalar field values to assign to the grid points.
         """
 
+        n_rows, n_cols = self.r.shape
         ugrid = vtk.vtkUnstructuredGrid()
+        ugrid.SetPoints(vtk_points)
         ugrid.GetPointData().SetScalars(vtk_scalars)
 
-        num_points = vtk_points.GetNumberOfPoints()
         cells = vtk.vtkCellArray()
 
-        for i in range(num_points):
-            vertex = vtk.vtkVertex()
-            vertex.GetPointIds().SetId(0, i)
-            cells.InsertNextCell(vertex)
+        def get_index(i, j):
+            return i * n_cols + j
 
-        ugrid = vtk.vtkUnstructuredGrid()
-        ugrid.SetPoints(vtk_points)
-        ugrid.GetPointData().SetScalars(vtk_scalars)
-        ugrid.SetCells(vtk.VTK_VERTEX, cells)
+        for i in range(n_rows - 1):
+            for j in range(n_cols - 1):
+                quad = vtk.vtkQuad()
+                quad.GetPointIds().SetId(0, get_index(i, j))
+                quad.GetPointIds().SetId(1, get_index(i, j + 1))
+                quad.GetPointIds().SetId(2, get_index(i + 1, j + 1))
+                quad.GetPointIds().SetId(3, get_index(i + 1, j))
+                cells.InsertNextCell(quad)
```

Comment:
This is actually pretty slow for larger grids (256 x 512 profiles_2d would already take a second on my laptop).

Can we do something smarter with numpy, like the following (not tested though 😉 )?
```suggestion
        quad = np.array([[0, 1, n_cols, n_cols + 1]])
        all_quads = quad + np.arange((n_cols - 1) * (n_rows - 1))[:, None]
        cells.SetCells(all_quads.shape[0], numpy_to_vtk(all_quads, deep=True, array_type=vtk.VTK_ID_TYPE))
```

pretty similar to how [the jorek reader populates the bezier cells](https://github.com/iterorganization/IMAS-ParaView/blob/develop/imas_paraview/io/read_jorek.py#L256)

Reply:
Good call! Your implementation gave me segfaults upon loading, but I did manage to get it working. Loading my test grid I have an improvement of ~40x (0.8s vs 0.01s). Let me know what you think!

Reply:
Very nice, thanks! 😄